src/library_proxyfs.js

```diff
@@ -192,6 +192,35 @@ mergeInto(LibraryManager.library, {
           throw new FS.ErrnoError(ERRNO_CODES[e.code]);
         }
       },
+      // NB: can't delegate mmap call from PROXYFS to underlying FS,
+      // because it may originate from another module,
+      // and then it'll use mmapAlloc of that module
+      // so the memory will be filled with file in that module, not in our module
+      // Instead, we should allocate module here on our own and fill it with the content
+      mmap: function(stream, length, position, prot, flags) {
+        var path = PROXYFS.realPath(stream.node);
+        var contents = stream.node.mount.opts.fs.readFile(path);
+
+        if (!(flags & 2)) {
+          throw new Error("Not implemented MAP_SHARED (yet)");
+        } else {
+          // Try to avoid unnecessary slices.
+          if (position > 0 || position + length < contents.length) {
+            if (contents.subarray) {
```

Comment:
When would `contents.subarray` not be available?

Reply:
Also not sure, I copied that from the existing mmap.

Guess, the same meaning as there =)

Reply:
It would be good to find out. Can you try to figure out when it might be necessary and possibly add a test for this case?